### PR TITLE
login: suggest other browsers when browser is unsupported

### LINKED ISSUE #9438: Login: Improved message for unsupported browsers
Currently, unsupported browsers get a message about WebSockets. It should be improved to suggest a list of various "known-good" browsers.

Of the top of my head, I'd suggest mentioning "a recent version" of:

- Firefox
- Chrome
- Edge
- Safari

...and omitting specific version numbers.

### PR DESCRIPTION
**Pull request by garrett**
Fixes #9438

### CODE REVIEW

**Comment by garrett** [comment]
Depends on #9735.

**Comment by garrett** [comment]
Looks like this:

screenshot_2018-08-02 sunny 1

Perhaps I should style the links to be a slight blue and/or underline. They should be more clickable. See screenshots in #9735.

**Comment by martinpitt** [comment]
#9735 landed yesterday. Conflicts, needs a rebase

File: src/ws/login.js [hunk 1]
```diff
@@ -239,9 +242,6 @@
             brand("brand", "Cockpit");
         }
```
Comment (martinpitt) [inline src/ws/login.js]: So before you still saw the badge/branding, but no login form. With this change, what hides the login form for non-met requisites? I. e. I'm sure you tested this, but I don't understand why this code moved. Maybe explain in the commit message?

**Comment by martinpitt** [comment]
Also, "browswers" in the commit message :)

**Comment by garrett** [comment]
Rebased.

File: src/ws/login.js (hunk 1 above)
Comment (garrett) [reply src/ws/login.js]: The login screen gets extremely jumbled with that information there. If you're using an old/broken/unsupported browser, not much other information matters until the browser situation is fixed.

**Comment by garrett** [comment]
I had WIP changes locally stored (oops!), so I tested in various scenarios, finished up the changes, and force pushed with them included.

In this new set of changes, I re-added the branding, centered the branding, and made the entire error within the block.

Desktop:

screenshot_2018-08-15 sunny

Mobile:

screen shot 2018-08-15 at 16 12 21

**Comment by garrett** [comment]
>  Improve error message 

Adding "API" to the string would cause translation issues. It should probably be handled in another PR, if at all. Up to your discretion.

**Comment by martinpitt** [comment]
@garrett: There is an outstanding TODO item in the above list. Also, I would like to write a test case for this. Is that possible with just changing the [user agent](https://chromedevtools.github.io/devtools-protocol/1-3/Emulation#method-setUserAgentOverride)? Or in other words, can this be tested interactively somehow with Chrome?

**Comment by martinpitt** [comment]
Suggestion from @larskarlitski : Monkey-patch `window.WebSocket = undefined`

**Comment by martinpitt** [comment]
This removes `#login-fatal`, so at least [this test](https://fedorapeople.org/groups/cockpit/logs/pull-9781-20180815-143434-719e83fb-verify-fedora-28/log.html#60) needs to be adjusted.

**Comment by martinpitt** [comment]
This can indeed be tested like this:
```py
        m.start_cockpit()
        # pretend browser doesn't support a required capability
        b.cdp.invoke("Page.addScriptToEvaluateOnLoad", scriptSource="window.WebSocket = undefined;")
        b.open("/system")
        b.wait_present("#login-fatal-block")
```

**Comment by martinpitt** [comment]
I reworked this to keep the "login-fatal" bit as it is, for the existing messages that aren't related to bad browsers. Instead I added an entirely separate `#unsupported-browser` section, and changed *this* case of `fatal()` to show it. I also added a test.

This can also be tested easily interactively by renaming one of the requirements in src/ws/login.js to a bogus value, building, and reloading the login page. Make a symlink like this to test it from the build tree:

    /usr/share/cockpit/static/login.min.html -> /home/martin/upstream/cockpit/dist/static/login.min.html

**Comment by martinpitt** [comment]
Fixed test on rhel-7-5-distropkg.

**Comment by martinpitt** [comment]
This would make a nice addition to today's release, so flagging priority for now.

**Comment by martinpitt** [comment]
Up to date screenshot for the release notes:

cockpit-unsupported-browser

**Comment by mvollmer** [comment]
Code looks good to me.

I am not sure whether leaving out the "This browser is too old to run Cockpit" message is the right thing.  Without it, it might not be totally clear what the user is looking at.  Just an advertisement for a better world?  Maybe I am okay with compromising on security, reliability, and performance if that allows me to use Cockpit on my old tablet. :-)

**Pull request merged**